src/Workspaces/Core/Portable/CodeStyle/CodeStyleOptions.cs

```diff
@@ -323,6 +337,13 @@ private static string GetAccessibilityModifiersRequiredEditorConfigString(CodeSt
                 KeyValuePairUtil.Create("never_if_unnecessary", ParenthesesPreference.NeverIfUnnecessary),
             });
 
+        private static readonly BidirectionalMap<string, UnusedParametersPreference> s_unusedParametersPreferenceMap =
+            new BidirectionalMap<string, UnusedParametersPreference>(new[]
+            {
+                KeyValuePairUtil.Create("none", UnusedParametersPreference.None),
+                KeyValuePairUtil.Create("private", UnusedParametersPreference.PrivateMethods),
+                KeyValuePairUtil.Create("all", UnusedParametersPreference.AllMethods),
```

Comment:
i feel it tracks more closely to user expectation to have "public, non-public, and private" There's no way for me to say, for example, "i want to know about internal methods that do this, but not public".  "public" is often my true public surface area, and i don't want these sorts of messages since i'm doing something for some necessary reason.  but anythign on my internal surface area (which does'nt necessarily mean private), is fair game and i want to know if i'm doing something smelly.

Reply:
Thanks - I have added "All" and "Non-public" for the most common user scenarios. If we receive feedback that people prefer even more stricter configuration and like to pick and choose accessibilities, we can change this to accept a comma separated list. I'd prefer to avoid that purely because once we have an explicit list it is hard to convey if the accessibilities mean effective accessibilities or declared accessibilities.

---
In reply to: [230606469](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606469) [](ancestors = 230606469)

Reply:
Agreed.  that sounds great to me.  Thanks!